tests/playwright-test/golden.spec.ts

```diff
@@ -488,6 +488,54 @@ test('should write missing expectations with sanitized snapshot name', async ({r
   expect(data.toString()).toBe('Hello world');
 });
 
+test('should lookup snapshot with path', async ({ runInlineTest }) => {
+  const result = await runInlineTest({
+    ...files,
+    'a.spec.js-snapshots/test/path/snapshot.txt': `Hello world`,
+    'a.spec.js': `
+      const { test } = require('./helper');
+      test('is a test', ({}) => {
+        expect('Hello world').toMatchSnapshot('snapshot.txt', { path: 'test/path' });
+      });
+    `
+  });
+
+  expect(result.exitCode).toBe(0);
+});
+
+test('should set snapshot with path', async ({ runInlineTest }, testInfo) => {
+  const result = await runInlineTest({
+    ...files,
+    'a.spec.js': `
+      const { test } = require('./helper');
+      test('is a test', ({}) => {
+        expect('Hello world').toMatchSnapshot('snapshot.txt', { path: 'test/path' });
```

Comment:
Should we instead partially revert #7620? The problem we were facing there was `path` going outside of the snapshots directory.

If we make `toMatchSnapshot('test/path/snapshot.txt')` to place snapshot at `a.spec.js-snapshots/test/path/snapshot.txt` that should just work, shouldn't it? This will save on extra params and APIs.

Reply:
Yeah good call I didn't notice that was a recent change. See changes in 9b9b677.